packages/core/strapi/lib/types/core/attributes/utils.d.ts

```diff
@@ -110,7 +110,7 @@ export type GetRequiredKeys<TSchemaUID extends Common.UID.Schema> = Utils.Object
   { required: true }
 >;
 
-export type GetOptionalKeys<TSchemaUID extends Common.UID.Schema> = keyof Omit<
+export type GetOptionalKeys<TSchemaUID extends Common.UID.Schema> = Utils.Object.KeysNotBy<
   GetAll<TSchemaUID>,
-  GetRequiredKeys<TSchemaUID>
+  { required: true }
```

Comment:
We can use this instead, that might end up being more maintainable

```suggestion
  Attribute.Required
```

(you can also change for the `GetRequiredKeys`)